pkg/k8s/lookup.go

```diff
@@ -80,15 +109,10 @@ func (r *RealK8sAPI) GetNumTyphas(namespace, serviceName, portName string) (int,
 }
 
 func (r *RealK8sAPI) GetNumNodes() (int, error) {
-	clientSet, err := r.clientSet()
-	if err != nil {
-		return 0, err
-	}
-
-	noClient := clientSet.CoreV1().Nodes()
-	list, err := noClient.List(v1.ListOptions{})
+	// Use the indexer's local store to get the number of nodes.
+	indexer, err := r.nodeIndexer()
```

Comment:
Can we check if the indexer is in sync?  Might be best to return an error rather than proceed with out-of-date information.

Reply:
Right now I have that as part of `getIndexer`, but could move that check here if you'd like: 

https://github.com/projectcalico/typha/pull/366/files#diff-7935b74e983a0f121cb9c320beba5c0fR75